File: Areas.m
clc; clear all; close all;
pkg load database;
retry= true;
retry2= true;
h=0;
while retry
  try
    fprintf("\n Elija una opcion \n 1.Correr el programa \n 2.Mostrar Historial \n");
    m=(input('Elija:   '));
    if m>2||m==0|m<0
      fprintf("Elija una opcion correcta \n");
    endif
    if m==1
      while retry2
        try
           fprintf("A continuacion elija la figura a la cual necesitasacar el area\n");
           fprintf("Elija una opcion:\n1.Circulo\n2.Triangulo\n3.Cuadrado\n4.Rectangulo\n");
           r=input("Elija:   ");
           if r>4||r==0|r<0
             fprintf("Elija una opcion correcta \n");
           endif
           if r==1
             ra=input("Ingrese el radio del Circulo:\n");
             fig="Circulo";
             a= (ra^2)*pi ;
           endif
           if r==2
             ra=input("Ingrese  la base del Triangulo:\n");
             h= input("Ingrese la altura del triangulo:\n");
             fig="Triangulo";
             a=ra*h*1/2 ;
           endif
           if r==3
             ra=input("Ingrese  el lado del cuadrado:\n");
             h = ra;
             a=ra^2;
             fig="Cuadrado";
           endif
           if r==4
             ra=input("Ingrese  la base del rectangulo:\n");
             h=input("Ingrese  la altura del rectangulo \n");
             a=ra*h;
             fig="Rectangulo";
           endif
           fprintf("El Area de %s es de: %d \n", fig,a);
           params= cell(1,4);
           params{1,1}=fig;
           params{1,2}=ra;
           params{1,3}=h;
           params{1,4}=a;
           conn = pq_connect(setdbopts('dbname','HWP','host','localhost',
           'port','5432','user','postgres','password','123fgthg'));
           N=pq_exec_params(conn, "insert into areas(fig, BaOR, h, A) values($1,$2,$3,$4);",params); %insertar datos en la tabla
           retry2=false;
           retry=false;
        catch
          fprintf("A ingresado un valor erroneo vuelva a intentarlo %d \n");
          msg = lasterror.message;
          fprintf(msg);
        end_try_catch
      endwhile
    endif
    if m==2
      conn = pq_connect(setdbopts('dbname','HWP','host','localhost',
      'port','5432','user','postgres','password','123fgthg'));
      N=pq_exec_params(conn, "Select * from Areas;"); %insertar datos en la tabla
      disp(N)
      retry=false;
    endif
  catch
    fprintf("A ingresado un valor erroneo vuelva a intentarlo %d \n");
    msg = lasterror.message;
    fprintf(msg);
  end_try_catch
endwhile

File: Areas.py
import psycopg2
from tabulate import tabulate
import numpy as np
def  post(x,y,z,w):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "HWP"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO areas(fig, BaOR, h, A) VALUES (%s, %s, %s, %s);", (x, y, z, w))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "123fgthg",
            dbname = "HWP"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from areas;")
    print(tabulate(cursor, headers=["ID", "FIGURA", "BASE O RADIO", "ALTURA", "AREA"], tablefmt="fancy_grid", numalign ="center"))

print("AREA DE DIFERENTES FIGURAS GEOMETRICAS \n")
while True:
    try:
        print("Elija una opcion:\n1.Correr el Programa\n2.Mostrar el Historial\n3.Salir")
        b=int(input("Elija:\n"))
        if b>3 or b<=0:
            print("Valor ingresado fuera del rango del menu.\n")
        elif b==1:
            while True:
                try:
                    print("A continuacion elija la figura a la cual necesitasacar el area\n")
                    print("Elija una opcion:\n1.Circulo\n2.Triangulo\n3.Cuadrado\n4.Rectangulo")
                    r=int(input("Elija:\n"))
                    h=0
                    if r>4 or b<=0:
                        print("Valor ingresado fuera del rango del menu.\n")
                    if r==1:
                        ra=int(input("Ingrese el radio del Circulo:\n"))
                        fig="Circulo"
                        a=np.pi*pow(ra,2)
                    if r==2:
                        ra=int(input("Ingrese  la base del Triangulo:\n"))
                        h= int(input("Ingrese la altura del triangulo:\n"))
                        fig="Triangulo"
                        a=ra*h*1/2
                    if r==3:
                        ra=int(input("Ingrese  el lado del cuadrado:\n"))
                        h = ra
                        a= pow(ra,2)
                        fig="Cuadrado"
                    if r==4:
                        ra = int(input("Ingrese la base del rectangulo\n"))
                        h = int(input("Ingrese la altura del rectangulo\n"))
                        a=ra*h
                        fig="Rectangulo"
                    print("El Area del "+ fig+" es:"+str(a)+" \n")
                    post(fig, ra, h, a)
                    break 
                except Exception as e:
                    print("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n")
                    print(repr(e))
        elif b==2:
            ret()
        elif b==3:
            break

    except Exception as e:
        print("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n")
        print(repr(e))

exit(0)